extensions/datasource/common/src/main/java/io/quarkus/datasource/common/runtime/DatabaseKind.java

```diff
@@ -0,0 +1,114 @@
+package io.quarkus.datasource.common.runtime;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * We don't use an enum as we also need to support non built-in database kinds.
+ */
+public final class DatabaseKind {
+
+    public static final String DERBY = "derby";
+    public static final String H2 = "h2";
+    public static final String MARIADB = "mariadb";
+    public static final String MSSQL = "mssql";
+    public static final String MYSQL = "mysql";
+    public static final String POSTGRESQL = "postgresql";
+
+    private static final Map<String, String> ALIASES;
+
+    static {
+        Map<String, String> aliases = new HashMap<String, String>();
```

Comment:
Maybe add the initial capacity?

Reply:
Also, you can just use the `<>` operator to avoid the extra generic type definition

Reply:
It's not easy to have the size of the map given you can have several aliases. Better leave it as that.

Granted for the diamond operator.